Sources/Build/TestMetadata.swift

```diff
@@ -0,0 +1,162 @@
+    
+    // Now parse and generate files for each test module.
+    // First parse each module's tests to get the list of classes
+    // which we'll use to generate the LinuxMain.swift file later.
+    let metadata = try product
+        .modules
+        .flatMap{ $0 as? TestModule }
+        .flatMap { try generateLinuxTestManifests($0) }
+    
+    //TODO: Decide what to do when users already have
+    //the linux XCTestCaseProvider extension. 
+    //Ignore that file? Error out and ask them to remove it once and for all?
+    //I'd prefer the second, because it'll get everyone on the same page quickly.
+    
+    // With the test information, generate the LinuxMain file
+    let mainPath = try generateLinuxMain(product, metadata: metadata)
+    
+    return ProductTestMetadata(linuxMainPath: mainPath, product: product, metadata: metadata)
+}
+
+/// Updates the contents of and returns the path of LinuxMain.swift file.
+func generateLinuxMain(product: Product, metadata: [ModuleTestMetadata]) throws -> String {
+    
+    // Now get the LinuxMain.swift file's path
+    // HACK: To get a path to LinuxMain.swift, we just grab the
+    //       parent directory of the first test module we can find.
+    let firstTestModule = product.modules.flatMap{ $0 as? TestModule }.first!
+    let testDirectory = firstTestModule.sources.root.parentDirectory
+    let main = Path.join(testDirectory, "LinuxMain.swift")
+    
+    // Generate the LinuxMain.swift's contents.
+    try writeLinuxMain(metadata, path: main)
+    
+    return main
+}
+
+/// Returns a list of class names that are subclasses of XCTestCase
+func generateLinuxTestManifests(module: TestModule) throws -> ModuleTestMetadata? {
+    
+    let root = module.sources.root
+    let testManifestPath = Path.join(root, "LinuxTestManifest.swift")
+    
+    // Replace the String based parser with AST parser once it's ready
+    let parser: TestMetadataParser = StringTestMetadataParser()
+    
+    let classes = try module
+        .sources
+        .relativePaths
+        .filter { $0 != "LinuxTestManifest.swift" }
+        .map { Path.join(root, $0) }
+        .flatMap { try parser.parseTestClasses($0) }
+        .flatMap { $0 }
+    
+    guard classes.count > 0 else { return nil }
+    
+    let metadata = ModuleTestMetadata(module: module, testManifestPath: testManifestPath, dependencies: module.dependencies.map { $0.name }, classes: classes)
+    
+    //now generate the LinuxTestManifest.swift file
+    try writeLinuxTestManifest(metadata, path: testManifestPath)
+    
+    return metadata
+}
+
+func writeLinuxTestManifest(metadata: ModuleTestMetadata, path: String) throws {
+    
+    let file = try fopen(path, mode: .Write)
+    defer {
+        fclose(file)
+    }
+    
+    //imports
+    try fputs("import XCTest\n", file)
+    try metadata.dependencies.forEach {
+        try fputs("@testable import \($0)\n", file)
```

Comment:
Sorry if I'm missing something, but do we need to import dependencies at all here? As far as I understand this...

- ...we're writing a test manifest that will be part of the test package that declares all the test cases and methods, correct? So we can reference the test cases directly, without an import.
- ...the test cases and their test methods won't reference their dependencies. So we don't need to import them.

Am I missing something?